services/horizon/internal/ingest/contractevents/events.go

```diff
@@ -0,0 +1,376 @@
+	if len(topics) != 4 {
+		return errors.New("mint event requires 4 topics")
+	}
+
+	// Admin is not used. but needs to be parsed for SAC format correctness
+	_, err := parseAddress(topics[1])
+	if err != nil {
+		return fmt.Errorf("invalid admin address: %w", err)
+	}
+	to, err := parseAddress(topics[2])
+	if err != nil {
+		return fmt.Errorf("invalid to address: %w", err)
+	}
+	event.To = to
+	return nil
+}
+
+func parseMintEventFromTxMetaV4(topics xdr.ScVec, event *StellarAssetContractEvent) error {
+	// Format: ["mint", to addr, sep11 asset] - NO admin address in V4
+	if len(topics) != 3 {
+		return errors.New("mint event requires 3 topics")
+	}
+
+	to, err := parseAddress(topics[1])
+	if err != nil {
+		return fmt.Errorf("invalid to address: %w", err)
+	}
+	event.To = to
+	return nil
+}
+
+func parseClawbackEventFromTxMetaV3(topics xdr.ScVec, event *StellarAssetContractEvent) error {
+	// Format: ["clawback", admin addr, from addr, sep11 asset], i128 amount
+	if len(topics) != 4 {
+		return errors.New("clawback event requires 4 topics")
+	}
+
+	// Admin is not used. but needs to be parsed for SAC format correctness
+	_, err := parseAddress(topics[1])
+	if err != nil {
+		return fmt.Errorf("invalid admin address: %w", err)
+	}
+	from, err := parseAddress(topics[2])
+	if err != nil {
+		return fmt.Errorf("invalid from address: %w", err)
+	}
+	event.From = from
+	return nil
+}
+
+func parseClawbackEventFromTxMetaV4(topics xdr.ScVec, event *StellarAssetContractEvent) error {
+	// Format: ["clawback", from addr, sep11 asset] - NO admin address in V4
+	if len(topics) != 3 {
+		return errors.New("clawback event requires 3 topics")
+	}
+
+	from, err := parseAddress(topics[1])
+	if err != nil {
+		return fmt.Errorf("invalid from address: %w", err)
+	}
+	event.From = from
+	return nil
+}
+
+func parseSacEventFromTxMetaV3(event *xdr.ContractEvent, networkPassphrase string) (*StellarAssetContractEvent, error) {
+	eventType, asset, topics, data, err := parseCommonEventValidation(event, networkPassphrase)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse amount (V3 is always direct i128)
+	amount, ok := data.GetI128()
+	if !ok {
+		return nil, fmt.Errorf("invalid amount in event data: %v", data.String())
+	}
+
+	// Parse addresses based on event type
+	sacEvent := &StellarAssetContractEvent{
+		Type:   eventType,
+		Asset:  asset,
+		Amount: amount,
+	}
+
+	switch eventType {
+	case EventTypeTransfer:
+		if err := parseTransferEvent(topics, sacEvent); err != nil {
+			return nil, err
+		}
+	case EventTypeMint:
+		if err := parseMintEventFromTxMetaV3(topics, sacEvent); err != nil {
+			return nil, err
+		}
+	case EventTypeClawback:
+		if err := parseClawbackEventFromTxMetaV3(topics, sacEvent); err != nil {
+			return nil, err
+		}
+	case EventTypeBurn:
+		if err := parseBurnEvent(topics, sacEvent); err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("%w: %v", ErrEventUnsupported, eventType)
```

Comment:
suggestion:
```suggestion
		return nil, errors.Join(ErrEventUnsupported, fmt.Errorf("%v", eventType))
```

though it only really matters if someone wants to `errors.Is(err, ErrEventUnsupported)` elsewhere.

Reply:
erm, in the `errors.Join()`,  `fmt.Errorf("%v", eventType)` is not really indicating an error, is it?
As in, I am not joining 2 different kinds of errors. 
I am trying to add more context to the unsupported type of error by appending event type.
Hence the choice of `fmt.Errorf("%w:%v)`

It was my understanding that you would use errors.Join for something like this:
 `errors.Join(HasBadInput1Parameter, AlsoHasBadInput2Parameter,...)`

I am not particularly opionated on this, and am happy to make the change, just want to understand it, is all